recipes/emela/meta.yaml

```diff
@@ -0,0 +1,125 @@
+{% set name = "emela" %}
+{% set version = "1.0" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/gstagnit/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: 53a867616f7ae0b03928c5a9db4b1b04a6b29794c9492a96b19dfea64b1c856d
+  folder: source
+  patches:
+    - 0001-support-range-of-cmakes.patch
+
+build:
+  skip: true  # [win]
+  number: 0
+  run_exports:
+    - {{ pin_subpackage('emela', max_pin='x.x') }}
+  script:
+    - cmake -DCMAKE_INSTALL_PREFIX=$PREFIX -DWITH_LHAPDF=ON -S source -B build
+    - cmake -LH build
+    - cmake --build build --clean-first --parallel="${CPU_COUNT}"
+    - cmake --install build
+
+requirements:
+  build:
+    - {{ stdlib('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('fortran') }}
+    - cmake
+    - make
+  host:
+    - libboost-devel <1.85
+    - lhapdf
+  run:
+    - lhapdf
+
+test:
+  source_files:
+    - source/example
+  requires:
+    - {{ compiler('cxx') }}
+    - cmake
+    - make
+    - curl
+    - tar
+  commands:
+    - test -f $PREFIX/include/eMELA/version.h
+    - test -f $PREFIX/include/eMELA/bspdf.hh
+    - test -f $PREFIX/include/eMELA/analytics.hh
+    - test -f $PREFIX/include/eMELA/specialfunctions.hh
+    - test -f $PREFIX/include/eMELA/version.h.in
+    - test -f $PREFIX/include/eMELA/constants.hh
+    - test -f $PREFIX/include/eMELA/eMELA.hh
+    - test -f $PREFIX/include/eMELA/grid.hh
+    - test -f $PREFIX/include/eMELA/ilc500.hh
+    - test -f $PREFIX/include/eMELA/numerics.hh
+    - test -f $PREFIX/lib/libeMELA${SHLIB_EXT}
+    - test -f $PREFIX/bin/eMELA-config
+
+    - eMELA-config --help
+    - eMELA-config --prefix
+    - eMELA-config --incdir
+    - eMELA-config --libdir
+    - eMELA-config --data
+    - eMELA-config --cppflags
+    - eMELA-config --cxxflags
+    - eMELA-config --ldflags
+
+    - cd source/example
+    - $CXX CreateGrid.cc -o CreateGrid $CXXFLAGS $LDFLAGS $(eMELA-config --ldflags)
+    # Skip as extremely slow with 50000 grid points
+    # - ./CreateGrid NLL DELTA MSBAR
+    - test -f ./CreateGrid
+    - $CXX GetPdf.cc -o GetPdf $CXXFLAGS $LDFLAGS $(eMELA-config --ldflags)
+    - ./GetPdf
+
+    # Download the grids from the GitHub releases page
+    # https://github.com/gstagnit/eMELA/releases/tag/v1.0
+    - curl -sLO https://github.com/gstagnit/eMELA/releases/download/v1.0/grids.tar.gz
```

Comment:
@xhochy that file is 115 MB. My understanding is that all `test_files` and `source_files` get added to the package `info/test` metadata.

From the CI (or `pixi run build-linux linux64`) logs:

```
Inspecting artifacts
-----------------------------------
linux-64/emela-1.0-h1ce2cf5_0.conda
-----------------------------------
-- Size: 160.5KB
-- SHA256: cf3e7dbedcab49d93cb458d685ad62e485d8b010c0aebae45fbe36f51c3c500f
-- Contents:
?rwxrwxr-x 0/0       4162 2024-12-05 05:20:44 bin/eMELA-config 
?rw-rw-r-- 0/0        945 2022-08-16 13:54:16 include/eMELA/analytics.hh 
?rw-rw-r-- 0/0       2796 2022-08-16 13:54:16 include/eMELA/bspdf.hh 
?rw-rw-r-- 0/0        227 2022-08-16 13:54:16 include/eMELA/constants.hh 
?rw-rw-r-- 0/0       7571 2022-08-16 13:54:16 include/eMELA/eMELA.hh 
?rw-rw-r-- 0/0        635 2022-08-16 13:54:16 include/eMELA/grid.hh 
?rw-rw-r-- 0/0       1184 2022-08-16 13:54:16 include/eMELA/ilc500.hh 
?rw-rw-r-- 0/0        232 2022-08-16 13:54:16 include/eMELA/numerics.hh 
?rw-rw-r-- 0/0       2172 2022-08-16 13:54:16 include/eMELA/specialfunctions.hh 
?rw-rw-r-- 0/0         22 2024-12-05 05:20:44 include/eMELA/version.h 
?rw-rw-r-- 0/0         34 2022-08-16 13:54:16 include/eMELA/version.h.in 
?rw-r--r-- 0/0       7673 2024-12-05 05:20:58 info/about.json 
?rw-r--r-- 0/0        290 2024-12-05 05:20:58 info/files 
?rw-r--r-- 0/0          0 2024-12-05 05:20:58 info/git 
?rw-r--r-- 0/0        278 2024-12-05 05:20:58 info/has_prefix 
?rw-r--r-- 0/0        312 2024-12-05 05:20:58 info/hash_input.json 
?rw-r--r-- 0/0        418 2024-12-05 05:20:58 info/index.json 
?rw-rw-r-- 0/0       1106 2022-08-16 13:54:16 info/licenses/source/LICENSE 
?rw-r--r-- 0/0       2727 2024-12-05 05:20:58 info/paths.json 
?rw-r--r-- 0/0       2294 2024-12-05 05:17:35 info/recipe/0001-support-range-of-cmakes.patch 
?rw-r--r-- 0/0       7256 2024-12-05 05:20:58 info/recipe/conda_build_config.yaml 
?rw-r--r-- 0/0       7299 2024-12-05 05:20:58 info/recipe/meta.yaml 
?rw-r--r-- 0/0       4463 2024-12-05 05:17:35 info/recipe/meta.yaml.template 
?rw-r--r-- 0/0         34 2024-12-05 05:20:58 info/run_exports.json 
?rw-r--r-- 0/0       1196 2024-12-05 05:20:58 info/test/run_test.sh 
?rw-rw-r-- 0/0        350 2022-08-16 13:54:16 info/test/source/example/CMakeLists.txt 
?rwxr-xr-x 0/0      22096 2024-12-05 05:20:53 info/test/source/example/CreateGrid 
?rw-rw-r-- 0/0        922 2022-08-16 13:54:16 info/test/source/example/CreateGrid.cc 
```

This would make the package file huge and these test files would not ever be used in a production setting. Is my understanding of this incorrect?

Reply:
Ah! I haven't thought of that.